lib/fluent/plugin/out_kinesis_producer.rb

```diff
@@ -20,6 +20,13 @@ class KinesisProducerOutput < BufferedOutput
     Fluent::Plugin.register_output('kinesis_producer', self)
     config_param_for_producer
 
+    def configure(conf)
+      super
+      unless @stream_name or @stream_name_prefix
+        raise Fluent::ConfigError, "'stream_name' or 'stream_name_prefix' is required"
```

Comment:
How should we handle if user specifies both `stream_name` and `stream_name_prefix`?